### ISSUE TITLE
"No such schema org.gnome.nautilus.preferences"

### ISSUE THREAD

**Issue by hydrargyrum** [GLolol/gnome-exe-thumbnailer#8]
I try to run exe-thumbnailer from CLI, it always spits me this confusing error message and does not thumbnail anything: `No such schema ?org.gnome.nautilus.preferences?`

**Comment by jlu5** [comment]
exe-thumbnailer isn't really meant to be run from the command line; it should instead be called through a supporting file manager. The rough set of arguments I outlined is as follows: `exe-thumbnailer <inputfile> <outputfile> <thumbnail URI>`, but that means that any freeform output paths not meeting the [FreeDesktop specification](https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html#DIRECTORY) won't show up in an FM.

The "no such schema" error is a harmless warning when Nautilus isn't installed, as the script tries to look up Nautilus' setting for max file size: https://github.com/exe-thumbnailer/exe-thumbnailer/blob/master/usr/bin/exe-thumbnailer#L206-L207. Perhaps the script can just suppress any STDERR output from that call if it fails.

**Comment by hydrargyrum** [comment]
Actually, it didn't generate anything, neither in the `outfile` (second param), neither in `~/.cache/thumbnails/*`.

**Comment by jlu5** [comment]
What full command line are you using?

**Comment by hydrargyrum** [comment]
After running it through `bash -x`, I believe the problem is with my input file which is considered too large for exe-thumbnailer (~472MB, a game installer), even though no error message indicates it clearly:

```
+ THUMBNAIL_LIMIT=
+ [[ ! -n '' ]]
+ THUMBNAIL_LIMIT=10485760
+ [[ exe = \m\s\i ]]
+ '[' 472610688 -lt 10485760 ']'
+ '[' '' ']'
+ exit 1
```

**Comment by jlu5** [comment]
Yeah, thumbnail generation is indeed capped at a certain amount to prevent the thumbnailer from eating up RAM/CPU. There's no console output because the app is intended to be silent (otherwise it'd be flooding ~/.xsession-errors and possibly syslog with messages)

So, I can break this down into smaller, specific tasks to implement:
- Suppress the "no such schema" warning from gsettings (this should be trivial)
- Possibly add a verbose logging flag
- Make the command line interface more clear (e.g. add --help/--version and better hints)

**Comment by hydrargyrum** [comment]
> Yeah, thumbnail generation is indeed capped at a certain amount to prevent the thumbnailer from eating up RAM/CPU.

I know absolutely nothing about exe format, but isn't retrieving the icon from an exe almost constant time?

> So, I can break this down into smaller, specific tasks to implement:

If I can add one feature request: take an optional argument for the size of the output thumbnail (i.e. `-s 256` for building a thumbnail fitting in 256x256, as some gnome thumbnailers do take).
(My goal is to add an exe-thumbnailer wrapper in my thumbnailing library project: https://github.com/hydrargyrum/vignette)

**Comment by jlu5** [comment]
> I know absolutely nothing about exe format, but isn't retrieving the icon from an exe almost constant time?

Not with the toolchain we're using, see https://bugs.launchpad.net/ubuntu/+source/gnome-exe-thumbnailer/+bug/602767 and https://bugs.launchpad.net/ubuntu/+source/icoutils/+bug/614918

> If I can add one feature request: take an optional argument for the size of the output thumbnail (i.e. -s 256 for building a thumbnail fitting in 256x256, as some gnome thumbnailers do take).

This is #3, but that's blocked because we don't have [any icon templates](https://github.com/exe-thumbnailer/exe-thumbnailer/tree/master/usr/share/pixmaps/exe-thumbnailer) that are bigger than 32x32 right now.

**Comment by hydrargyrum** [comment]
@GLolol Why did you close the issue? What's the status on it?

**Comment by jlu5** [comment]
This is odd, I honestly thought I fixed this issue in 0.10 already...
